vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -175,16 +178,26 @@ def __init__(self,
         ops.register_buffer(self._ptr, self.buffer_ptrs)
 
     @staticmethod
-    def create_shared_buffer(
-            size_in_bytes: int,
-            group: Optional[ProcessGroup] = None) -> List[int]:
+    def create_shared_buffer(size_in_bytes: int,
+                             group: Optional[ProcessGroup] = None,
+                             uncached: Optional[bool] = False) -> List[int]:
         """
         Creates a shared buffer and returns a list of pointers
         representing the buffer on all processes in the group.
+        uncached - meta data buffers need to be "uncached" for signal on MI200
         """
-        lib = CudaRTLibrary()
-        pointer = lib.cudaMalloc(size_in_bytes)
-        handle = lib.cudaIpcGetMemHandle(pointer)
+        if current_platform.is_rocm():
```

Comment:
All of this platform specific dispatching logic does make me think that we may be close to wanting to split CustomAllReduce into platform specific subclasses.